### PR TITLE
add project validation to project client (#1335)

### PR DESCRIPTION
**Pull request by jsilverio22**
### Checklist
- [x] The title of this PR would make a good line in Acorn's Release Note's Changelog
- [x] The title of this PR ends with a link to the main issue being address in paranthesis, like: `This is a title (#1216)`. [Here's an example](https://github.com/acorn-io/acorn/pull/1199)
- [x] All relevant issues are referenced in the PR description. *NOTE: don't use [GitHub keyworkds](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) that auto-close issues*
#1335 
- [x] Commits follow [contributing guidance](https://github.com/acorn-io/acorn/blob/main/CONTRIBUTING.md#commits)
- [x] Automated tests added to cover the changes. If tests couldn't be added, an explanation is provided in the Verification and Testing section
- [ ] Changes to user-facing functionality, API, CLI, and upgrade impacts are clearly called out in PR description
- [ ] PR has at least two approvals before merging (or a reasonable exception, like it's just a docs change)

### CODE REVIEW

**Review by yashgiri — APPROVED** [review]
LGTM

File: pkg/project/client.go [hunk 1]
```diff
@@ -213,6 +215,9 @@ func getDesiredProjects(ctx context.Context, cfg *config.CLIConfig, opts Options
 	if strings.TrimSpace(p) == "" {
 		return nil, nil
 	}
+	if !opts.Create && len(projects) != 0 && !slices.Contains(projects, p) {
```
Comment (thedadams) [inline pkg/project/client.go:218]: nit: can this be simplified?
```suggestion
	if !opts.Create && !slices.Contains(projects, p) {
```

**Review by thedadams — APPROVED** [review]

File: pkg/project/client.go (hunk 1 above)
Comment (jsilverio22) [reply pkg/project/client.go]: yep had it in there before the opts.Create but no longer need now

File: pkg/project/client.go [hunk 2]
```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/acorn-io/acorn/pkg/hub"
 	"github.com/acorn-io/acorn/pkg/system"
 	"github.com/acorn-io/baaah/pkg/restconfig"
+	"golang.org/x/exp/slices"
```
Comment (cjellick) [inline pkg/project/client.go:15]: we pretty consistently use the `"k8s.io/utils/strings/slices"` library

**Pull request merged**